src/jit/morph.cpp

```diff
@@ -7017,24 +7164,83 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
             {
 #if defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
 
-                unsigned typeSize = 0;
-                hasMultiByteArgs  = !VarTypeIsMultiByteAndCanEnreg(argx->TypeGet(), objClass, &typeSize, false);
+                // hasMultiByteStackArgs will determine if the struct can be passed
+                // in registers. If it cannot we will break the loop and not
+                // fastTailCall. See https://github.com/dotnet/coreclr/issues/12644.
+                unsigned typeSize     = 0;
```

Comment:
Please make sure the entire description is here in the comments, and the reader does not need to refer to GitHub (or any other link) to understand the issue.